apps/emqx_rule_engine/src/emqx_rule_funcs.erl

```diff
@@ -1167,3 +1170,25 @@ convert_timestamp(MillisecondsTimestamp) ->
     Secs = MicroTimestamp div 1000_000 - MegaSecs * 1000_000,
     MicroSecs = MicroTimestamp rem 1000_000,
     {MegaSecs, Secs, MicroSecs}.
+
+%% @doc This currently undisclosed feature facilitates experimentation
+%% with various impromptu concepts.
+%% While it enables the integration of third-party plugin functions,
+%% it might also introduce unforeseen consequences.
+%% Caution is advised when utilizing this feature!
+%% @doc
+%% e.g.
+%% -module(my_module).
+%% -epxort([is_my_topic/1]).
+%% %% return true if the topic has 5 levels
+%% is_my_topic(Topic) ->
+%%   emqx_topic:levels(Topic) =:= 5.
+%%
+%% Use it in SQL:
+%% SELECT * FROM t/# WHERE eval('my_module:is_my_topic', topic)
+%%
+eval(ModFun, Args) ->
```

Comment:
The user may want to pass variable number of args to the customized function. See the implementation of the `sprintf` function at: https://github.com/emqx/emqx/blob/0617a9b11c227a70e48c6da9811ece60274f0322/apps/emqx_rule_engine/src/emqx_rule_funcs.erl#LL1124C12-L1124C12

Reply:
I'm not convinced that this is preferable in this case:

```
eval('mymod:myfun', something, 34324, 24)
eval('mymod:myfun', [something, 34324, 24])
```

I think it is more clear in the second example what the list of parameters to myfun is.